flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/data/binary/BinaryStringDataUtil.java

```diff
@@ -144,12 +151,16 @@
         return substrings.toArray(new BinaryStringData[0]);
     }
 
-    /** Decide boolean representation of a string. */
-    public static Boolean toBooleanSQL(BinaryStringData str) {
+    /** Parse a {@link StringData} to boolean. */
+    public static boolean toBoolean(BinaryStringData str) throws TableException {
```

Comment:
move the changes in `BinaryStringDataUtil` in a separate commit to be visible in the log

Reply:
I can't easily extract the changes of `BinaryStringDataUtil` without creating commits that won't build, I renamed the commit where the big changes happen